src/compile.rs

```diff
@@ -682,6 +706,145 @@ fn check_required_config<'a>(
     ret.into_result(())
 }
 
+pub fn get_infos<'a>(
+    new_type_map: &TypeMap,
+    policies: &'a [PolicyFile],
+    classlist: &ClassList,
+    system_configs: &Option<&BTreeMap<String, &'a Argument>>,
+) -> Result<Vec<sexp::Sexp>, CascadeErrors> {
+    let ret = CascadeErrors::new();
+    // Get the policy files
+    let new_policies = get_policy_files(new_type_map, policies);
+
+    // Get the function infos
+    let mut new_func_map = get_funcs(&new_policies, new_type_map)?;
+
+    // Validate functions
+    let new_func_map_copy = new_func_map.clone(); // In order to read function info while mutating
+    validate_functions(
+        &mut new_func_map,
+        new_type_map,
+        classlist,
+        &new_func_map_copy,
+    )?;
+
+    // Get the policy rules
+    let new_policy_rules = get_policy_rules(&new_policies, new_type_map, classlist, &new_func_map)?;
+
+    let new_cil_tree = generate_sexp(
+        new_type_map,
+        classlist,
+        new_policy_rules,
+        &new_func_map,
+        system_configs,
+    )?;
+
+    ret.into_result(new_cil_tree)
+}
+
+pub fn get_reduced_types(
```

Comment:
This could use a comment that goes over the rules for reducing types, notably the fact that we include parents automatically.

It's probably also worth mentioning that we clone the types, because ownership.